account_move_tier_validation/tests/test_tier_validation.py

```diff
@@ -66,25 +69,25 @@ def test_03_move_post(self):
                 "reviewer_id": self.test_user_1.id,
             }
         )
+
         partner = self.env["res.partner"].create({"name": "Test Partner"})
         product = self.env["product.product"].create({"name": "Test product"})
+
         invoice = self.env["account.move"].create(
             {
                 "move_type": "out_invoice",
                 "partner_id": partner.id,
-                "invoice_date_due": fields.Date.from_string("2024-01-01"),
+                "invoice_date_due": fields.Date.to_date("2024-01-01"),
                 "invoice_line_ids": [
                     (0, 0, {"product_id": product.id, "quantity": 1, "price_unit": 30})
                 ],
             }
         )
-        invoice.with_user(self.test_user_2.id).request_validation()
+        reviews = invoice.with_user(self.test_user_2.id).request_validation()
+        self.assertTrue(reviews)
+        with self.assertRaises(ValidationError):
```

Comment:
Please use assertRaisesRegex (plus savepoint) to give an indication which error occurs here.

Reply:
Hi @StefanRijnhart 

Could we write test like this?
```
with self.assertRaisesRegex(ValidationError, "You cannot post an invoice that requires tier validation"):
            invoice.action_post()
```

PS. IMHO we do not need to skip_validation_check because it will skip the error so this ensure if user try to post invoice while it on validation process, it should raise some error.

@aarosales-spearhead 
Can you try with this?

Reply:
@TheerayutEncoder Yes, that's how assertRaisesRegex can be used. You can also replace some of the text with `.*`.

Your humble opinion aside :smile:, I just tested without passing the context and then `action_post()` fails on `You are not allowed to write those fields under validation.`. The original test code remains valid and should be restored.